samples/transactional-session/aspnetcore-webapi/SqlPersistenceTS_6/WebApplication/MessageSessionFilter.cs

```diff
@@ -14,9 +14,12 @@ public async Task OnResourceExecutionAsync(ResourceExecutingContext context, Res
             var session = context.HttpContext.RequestServices.GetRequiredService<ITransactionalSession>();
             await session.Open(new SqlPersistenceOpenSessionOptions());
 
-            await next();
+            var result = await next();
 
-            await session.Commit();
+            if(result.Exception == null)
```

Comment:
Don't remember what LangVersion we use these days but if you want you could the `is null` pattern.

Reply:
good idea!